besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2235,6 +2237,17 @@ private DataStorageConfiguration getDataStorageConfiguration() {
     if (dataStorageConfiguration == null) {
       dataStorageConfiguration = dataStorageOptions.toDomainObject();
     }
+    if (BONSAI.equals(dataStorageConfiguration.getDataStorageFormat())
+        && dataStorageConfiguration.getBonsaiLimitTrieLogsEnabled()
+        && SyncMode.FULL.equals(getDefaultSyncModeIfNotSet())) {
```

Comment:
That is intended. The alternative is to automatically set bonsai-limit-trie-logs-enabled to false which you commented on here [`0726278` (#7140)](https://github.com/hyperledger/besu/pull/7140/commits/07262789162c1d3d3a637ed0c52ffc536e88ffe0#r1655487660) and has been discussed in previous threads. 
I am reconsidering the auto-set at the moment instead of this PR.

Reply:
Here's the alternative PR to auto-set to false: https://github.com/hyperledger/besu/pull/7357